### DISCUSSION TITLE
Multilanguage/localisation

### DISCUSSION THREAD

**Discussion by rossmcphee** [oqtane/oqtane.framework#734] [General] [answered]
I'm trying it add multilanguage support to an external module (namespace Rtv.Local4).

In the Rtv.Local4.Client project I've created a LocalisedText.resx containing my default English text and a LocalisedText.DE.resx with the German translations.  This produces an extra DLL named Rtv.Local4..Client.Oqtane.resources.dll, in a new folder called DE below the Client project's netcoreapp3.1 folder. So far so expected.

I've amended the .cmd files in the Rtv.Local4.Package project to copy the extra Rtv.Local4..Client.Oqtane.resources.dll file into a DE folder below the Octane Servers netcoreapp3.1 and was then hoping that I could access the translations from the client's Razor pages thus:

```
        Thread.CurrentThread.CurrentCulture = new System.Globalization.CultureInfo("de-DE"); ;
        Thread.CurrentThread.CurrentUICulture = new System.Globalization.CultureInfo("de-DE");
        LoadingStr = Rtv.Local4.Client.LocalisedText.Index_Loading;
```

But that's not working - the English text continues to appear regardless of the current culture.

I wrote a little Console app to test the Rtv.Local4.Client and this does work as expected:
```
        static void Main(string[] args)
        {
            Console.WriteLine(Rtv.Local4.Client.LocalisedText.Index_Loading);
            Thread.CurrentThread.CurrentCulture = new System.Globalization.CultureInfo("de-DE"); ;
            Thread.CurrentThread.CurrentUICulture = new System.Globalization.CultureInfo("de-DE");
            Console.WriteLine(Rtv.Local4.Client.LocalisedText.Index_Loading);
        }
```
Produces:
```
	Loading
	Laden
```
Has anyone got localisation working successfully in Oqtane?

I notice there a mention of "Static Localization" on the road map - is that just for Oqtane Server or are changes being considered for external modules as well?

**Comment by hishamco** [comment]
Did you use the localization APIs? BTW the localization on the roadmap AFAIK, I'm glad to work on this in the upcoming future ...

**Reply by rossmcphee** [reply]
Which localisation APIs do you mean?  I started hacking around with the Blazor IStringLocalizer but stopped when I realised I couldn't inject it into an external module.  After that I tried doing it 'old school' with resx files as described above.

**Reply by hishamco** [reply]
> Which localisation APIs do you mean?

ASP.NET Core Localization APIs

> I started hacking around with the Blazor IStringLocalizer but stopped when I realised I couldn't inject it into an external module

Did you add `AddLocalization` in your `Program.cs`?

**Comment by sbwalker** [comment]
@rossmcphee the static localization enhancement on the roadmap will need to work for all module scenarios - whether they are internal to the Oqtane framework or in external modules.

**Comment by sbwalker** [comment, marked as answer]
@rossmcphee see #749 - there is a discussion on adding Localization support to the next major version of Oqtane
